source/base/parameter_acceptor.cc

```diff
@@ -0,0 +1,241 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2017 by the deal.II authors
+//
+//    This file is part of the deal.II library.
+//
+//    The deal.II library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE at
+//    the top level of the deal.II distribution.
+//
+//-----------------------------------------------------------
+
+#include <deal.II/base/parameter_acceptor.h>
+#include <deal.II/base/utilities.h>
+#include <deal.II/base/revision.h>
+#include <deal.II/base/path_search.h>
+#include <boost/core/demangle.hpp>
+#include <fstream>
+
+DEAL_II_NAMESPACE_OPEN
+
+
+// Static empty class list
+std::vector<SmartPointer<ParameterAcceptor> > ParameterAcceptor::class_list;
+// Static parameter handler
+ParameterHandler ParameterAcceptor::prm;
+
+ParameterAcceptor::ParameterAcceptor(const std::string name) :
+  acceptor_id(class_list.size()),
+  section_name(name)
+{
+  SmartPointer<ParameterAcceptor> pt(this, boost::core::demangle(typeid(*this).name()).c_str());
+  class_list.push_back(pt);
+}
+
+
+ParameterAcceptor::~ParameterAcceptor()
+{
+  class_list[acceptor_id] = 0;
+}
+
+std::string ParameterAcceptor::get_section_name() const
+{
+  return (section_name != "" ? section_name : boost::core::demangle(typeid(*this).name()));
+}
+
+
+void
+ParameterAcceptor::initialize(const std::string filename,
+                              const std::string output_filename,
+                              const ParameterHandler::OutputStyle output_style_for_prm_format)
+{
+  declare_all_parameters();
+  if (filename != "")
+    {
+      // check the extension of input file
+      if (filename.substr(filename.find_last_of(".") + 1) == "prm")
+        {
+          try
+            {
+              prm.parse_input(filename);
+            }
+          catch (dealii::PathSearch::ExcFileNotFound)
+            {
+              std::ofstream out(filename);
+              Assert(out, ExcIO());
+              prm.print_parameters(out, ParameterHandler::Text);
+              out.close();
+              AssertThrow(false, ExcMessage("You specified "+filename+" as input "+
```

Comment:
can you write this as `"You specified <"+filename+"> as input "+`? This seems to be how we do this elsewhere